### PR TITLE
TXAD-19: Implement permit list APIs

### PR DESCRIPTION
**Pull request by objectorientguy**

### CODE REVIEW

File: lib/screens/permit/permit_details_screen.dart [hunk 1]
```diff
@@ -61,10 +60,10 @@ class PermitDetailsScreen extends StatelessWidget {
                     ],
                   ),
                 ),
-                subtitle: const Padding(
-                  padding: EdgeInsets.only(top: midTiniestSpacing),
-                  child: Tags(),
-                ),
+                // subtitle: const Padding(
```
Comment (Aditi-005) [inline lib/screens/permit/permit_details_screen.dart:63]: Remove commented code

File: lib/screens/permit/widgets/permit_details.dart [hunk 2]
```diff
@@ -22,7 +21,7 @@ class PermitDetails extends StatelessWidget {
               .copyWith(color: AppColor.black, fontWeight: FontWeight.bold),
         ),
         const SizedBox(height: midTiniestSpacing),
-        const DateTimeRow(),
+        // const DateTimeRow(),
```
Comment (Aditi-005) [inline lib/screens/permit/widgets/permit_details.dart:24]: Remove commented code

File: lib/widgets/icon_and_text_row.dart [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import 'package:flutter/material.dart';
+
+import '../configs/app_dimensions.dart';
+import '../configs/app_spacing.dart';
+
+class IconAndTextRow extends StatelessWidget {
+  final String? title;
```
Comment (Aditi-005) [inline lib/widgets/icon_and_text_row.dart:7]: no need to make it nullable

File: lib/widgets/icon_and_text_row.dart [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  const IconAndTextRow({super.key, this.title = '', this.icon});
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Image.asset('assets/icons/$icon.png',
+            height: kImageHeight, width: kImageWidth),
+        const SizedBox(width: tiniestSpacing),
+        Text(title!)
```
Comment (Aditi-005) [inline lib/widgets/icon_and_text_row.dart:19]: remove null check

**Review by Aditi-005 — CHANGES_REQUESTED** [review]
Please make the requested changes

File: lib/data/models/status_tag_model.dart [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import 'package:flutter/material.dart';
+
+class StatusTagModel {
+  final String? title;
```
[the remark this answers is not in the document]
Comment (objectorientguy) [reply lib/data/models/status_tag_model.dart:4]: done

File: lib/screens/permit/widgets/permit_details.dart (hunk 2 above)
Comment (objectorientguy) [reply lib/screens/permit/widgets/permit_details.dart:24]: done

File: lib/widgets/icon_and_text_row.dart (hunk 4 above)
Comment (objectorientguy) [reply lib/widgets/icon_and_text_row.dart:19]: why?

File: lib/widgets/icon_and_text_row.dart (hunk 3 above)
Comment (objectorientguy) [reply lib/widgets/icon_and_text_row.dart:7]: why?

Comment (Aditi-005) [reply lib/widgets/icon_and_text_row.dart:7]: If you have already assigned initial value as "" then it won't ever be "null".  Its basically not in use at all, and you had to give an unnecessary null check

File: lib/widgets/icon_and_text_row.dart (hunk 4 above)
Comment (Aditi-005) [reply lib/widgets/icon_and_text_row.dart:19]: If you have already assigned initial value as "" then it won't ever be "null". Its an unnecessary null check

**Review by Aditi-005 — APPROVED** [review]
Approved

**Pull request merged**